### PR TITLE
test(*): add a11y tests and some a11y fix

### PR DESCRIPTION
**Pull request by guanpu**
- Add accessibility tests for Tab, Range, Switch, Radio, Checkbox, Grid
- Fix some accessibility issues found during the tests
- Add a11y related documentation for Grid

### CODE REVIEW

File: src/tab/tabs/nav.jsx [hunk 1]
```diff
@@ -424,21 +527,40 @@ class Nav extends React.Component {
...
+                                className={navCls}
+                                animation={navCls}
+                                singleMode={false}
+                                ref={this.navRefHandler}
+                            >
+                                {tabList}
+                            </Animate>
+                        ) : (
+                            <ul
+                                role="tablist"
```
Comment (guanpu) [inline src/tab/tabs/nav.jsx]: This line is the only real change, I mean add `role="tablist"`, others are all prettier thing.

File: test/grid/a11y-spec.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import { unmount, testReact } from '../util/a11y/validate';
+
+/* eslint-disable react/jsx-filename-extension */
+/* global describe it beforeEach afterEach */
+
+const { Row, Col } = Grid;
+
+Enzyme.configure({ adapter: new Adapter() });
+
+describe('Row', () => {
```
Comment (jdkahn) [inline test/grid/a11y-spec.js]: change to 'Row A11y' for consistency

File: test/grid/a11y-spec.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    afterEach(() => {
+        if (wrapper) {
+            wrapper.unmount();
+            wrapper = null;
+        }
+        unmount();
+    });
+
+    it('should simply pass', async () => {
```
Comment (jdkahn) [inline test/grid/a11y-spec.js]: change to 'should not have any violations'

File: test/range/a11y-spec.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import simulateEvent from 'simulate-event';
+import assert from 'power-assert';
+import sinon from 'sinon';
+import Range from '../../src/range/index';
+import { unmount, testReact } from '../util/a11y/validate';
+
+/* eslint-disable react/no-multi-comp */
+Enzyme.configure({ adapter: new Adapter() });
+
+describe('Range ', () => {
```
Comment (jdkahn) [inline test/range/a11y-spec.js]: 'Range A11y'

File: test/tab/a11y-spec.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    let wrapper;
+
+    afterEach(() => {
+        if (wrapper) {
+            wrapper.unmount();
+            wrapper = null;
+        }
+        unmount();
+    });
+    it('should should not report any issue for tab with children', async () => {
```
Comment (jdkahn) [inline test/tab/a11y-spec.js]: 'should should have any violations for tab with children'

File: test/tab/a11y-spec.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+        wrapper = await testReact(<Tab><Tab.Item title="foo" /></Tab>);
+        return wrapper;
+    });
+    it('should should not report any issue for tab with datasource', async () => {
```
Comment (jdkahn) [inline test/tab/a11y-spec.js]: 'should not have any violations for tab with datasource'

File: test/grid/a11y-spec.js (hunk 2 above)
Comment (guanpu) [reply test/grid/a11y-spec.js]: okay

File: test/grid/a11y-spec.js (hunk 3 above)
Comment (guanpu) [reply test/grid/a11y-spec.js]: Accepted

File: test/range/a11y-spec.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+        }
+        unmount();
+    });
+    it('should not have any violations for range without tips', async () => {
+        wrapper = await testReact(<div>
+            <Range defaultValue={10} hasTip={false} tooltipVisible={false} />
+        </div>);
+        return wrapper;
+    });
+    it.skip('should not have any violations for range with tips', async () => {
```
Comment (jdkahn) [inline test/range/a11y-spec.js]: can you add a comment for why this is skipped?

Comment (guanpu) [reply test/range/a11y-spec.js]: Done

**Comment by guanpu** [comment]
The coverage drop is due to Prettier re-formatting.

**Comment by youluna** [comment]
@guanpu merge master and resolve conflicts please~

**Pull request merged**